libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt

```diff
@@ -219,14 +228,28 @@ object MapboxDistanceUtil {
         }
     }
 
-    private fun getUnitString(resources: Resources, @TurfConstants.TurfUnitCriteria unit: String) =
-        when (unit) {
-            TurfConstants.UNIT_KILOMETERS -> resources.getString(R.string.mapbox_unit_kilometers)
-            TurfConstants.UNIT_METERS -> resources.getString(R.string.mapbox_unit_meters)
-            TurfConstants.UNIT_MILES -> resources.getString(R.string.mapbox_unit_miles)
-            TurfConstants.UNIT_FEET -> resources.getString(R.string.mapbox_unit_feet)
-            else -> ""
+    private fun getUnitString(
+        resources: Resources,
+        @TurfConstants.TurfUnitCriteria unit: String
+    ): String {
+        return try {
+            when (unit) {
+                TurfConstants.UNIT_KILOMETERS -> {
+                    resources.getString(R.string.mapbox_unit_kilometers)
+                }
+                TurfConstants.UNIT_METERS -> resources.getString(R.string.mapbox_unit_meters)
+                TurfConstants.UNIT_MILES -> resources.getString(R.string.mapbox_unit_miles)
+                TurfConstants.UNIT_FEET -> resources.getString(R.string.mapbox_unit_feet)
+                else -> ""
+            }
+        } catch (ex: Exception) {
```

Comment:
Just a suggestion: maybe catch Throwable? I once had an issue where instead of `ClassNotFoundException` `ClassNotFoundError` was thrown but I was catching `Exception`. If we don't expect any specific Exception type here, I'd go with Throwable.